### PR TITLE
Adding sample transport fallback

### PR DESCRIPTION
**Pull request by moozzyk**
/cc @BrennanConroy @mikaelm12 @davidfowl

### CODE REVIEW

File: ChatSample/ChatSample/Startup.cs [hunk 1]
```diff
@@ -29,7 +30,7 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
 
             app.UseSignalR(routes =>
             {
-                routes.MapHub<ChatHub>("chat");
+                routes.MapHub<ChatHub>("chat", options => { options.Transports = TransportType.All; });
```
Comment (davidfowl) [inline ChatSample/ChatSample/Startup.cs]: Why this change?

Comment (moozzyk) [reply ChatSample/ChatSample/Startup.cs]: I used this to test fallback and decided to leave to show that you can pass options to MapHub. I don't think we have this in any of the samples.

Comment (davidfowl) [reply ChatSample/ChatSample/Startup.cs]: We should remove it. It's ugly and not required. If we had a sample showing specifically how to only use support websockets, it would be more useful.

Comment (moozzyk) [reply ChatSample/ChatSample/Startup.cs]: OK

**Comment by moozzyk** [comment]
:up::date:

File: ChatSample/ChatSample/wwwroot/index.html [hunk 2]
```diff
@@ -62,7 +63,33 @@
                     messageInput.focus();
                     event.preventDefault();
                 });
+            })
+            .catch(error => {
+                console.error(error.message);
             });
+
+            function startConnection(url, configureConnection) {
```
Comment (davidfowl) [inline ChatSample/ChatSample/wwwroot/index.html]: Put a comment in here.

File: ChatSample/ChatSample/Startup.cs (hunk 1 above)
Comment (asfaya) [reply ChatSample/ChatSample/Startup.cs]: Do you any other transport fallback protocol sample?
I'm creating my first SignalR functionality on a .netcore + ng4 app and I would really appreciate any sample that can help me leverage the problem of determining an alternative transport protocol.
Regards

Comment (moozzyk) [reply ChatSample/ChatSample/Startup.cs]: @asfaya - why would you need a fallback for transport protocol? Currently, the server supports both MsgPack and JSon so, whichever you pick you will be fine.

Comment (asfaya) [reply ChatSample/ChatSample/Startup.cs]: While running the app from kestrel everything works fine.
If, from VS, I run the app on IIS Express, occurs and error because the client tries to use websockets and the server is not responding. And there is no "automatic fallback" to long pooling (for instance)
So, I was thinking in trying to catch the error when not being able to connect with websockets and try again using long pooling.

Comment (davidfowl) [reply ChatSample/ChatSample/Startup.cs]: Are you on windows 7?

Comment (asfaya) [reply ChatSample/ChatSample/Startup.cs]: Hi David, sorry for the delay. Yes, I'm using Win7 x64
VS 2017 + IIS Express 

Thanks!

Andrés

Comment (moozzyk) [reply ChatSample/ChatSample/Startup.cs]: If you know your server does not support webSockets (i.e. you are running behind IIS/IISExpress on Win7) you can disable the WebSockets transport on the server side. This is basically https://github.com/aspnet/SignalR/issues/1033

Comment (asfaya) [reply ChatSample/ChatSample/Startup.cs]: I'll check that out. Thank you!
